pkg/archive/archive.go

```diff
@@ -25,12 +25,17 @@ import (
 )
 
 type (
-	Archive         io.ReadCloser
-	ArchiveReader   io.Reader
-	Compression     int
+	// Archive is a type of io.ReadCloser which has two interfaces Read and Closer.
+	Archive io.ReadCloser
+	// ArchiveReader is a type of io.Reader.
+	ArchiveReader io.Reader
```

Comment:
`ArchiveReader` will cause a golint warnings `type name will be used as archive.ArchiveReader by other packages, and that stutters; consider calling this Reader`, how to rename `ArchiveReader`, how to rename `ArchiveReader`? Anyone has a suggestions?